ethers/providers/jsonrpc/subscriptions.nim

```diff
@@ -151,14 +152,24 @@ proc new*(_: type JsonRpcSubscriptions,
   proc getChanges(originalId: JsonNode): Future[JsonNode] {.async.} =
     try:
       let mappedId = subscriptions.subscriptionMapping[originalId]
-      return await subscriptions.client.eth_getFilterChanges(mappedId)
+      let changes = await subscriptions.client.eth_getFilterChanges(mappedId)
+      if changes.kind == JNull:
+        return newJArray()
+      elif changes.kind != JArray:
+        raise newException(SubscriptionError,
+          "HTTP polling: unexpected value returned from eth_getFilterChanges." &
+          " Expected: JArray, got: " & $changes.kind)
+      return changes
+    except CancelledError as e:
+      raise e
     except CatchableError as e:
       if "filter not found" in e.msg:
         let filter = subscriptions.filters[originalId]
         let newId = await subscriptions.client.eth_newFilter(filter)
         subscriptions.subscriptionMapping[originalId] = newId
```

Comment:
Btw. we are leaving a place for race condition here. While its probability is small, it can still happen. The `await` here yields the processing of this, so if meanwhile for example `unsubscribe` is called, then it will use the old ID because the new one was not yet processed. I will leave up to you if you want to mitigate that or will leave it as is.

Reply:
To be addressed in https://github.com/codex-storage/nim-ethers/issues/86.